frontend/src/metabase/hooks/use-on-outside-click.ts

```diff
@@ -0,0 +1,22 @@
+import { useEffect, RefObject } from "react";
+
+interface ValidRefTarget {
+  contains(target: EventTarget | null): boolean;
+}
+
+export function useOnOutsideClick<T extends ValidRefTarget = HTMLDivElement>(
```

Comment:
Maybe consider calling it `useOnClickOutside`.

Would be inline with the function name in it, `handleClickOutside`.

Also seems more expected.